programming-extensions/programming-extension-dataspaces/src/test/java/dataspaces/ProActiveFileObjectTest.java

```diff
@@ -0,0 +1,102 @@
+ * as published by the Free Software Foundation: version 3 of
+ * the License.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ * If needed, contact us to obtain a release under GPL Version 2 or 3
+ * or a different license than the AGPL.
+ */
+package dataspaces;
+
+import static org.junit.Assert.assertTrue;
+
+import java.io.File;
+import java.io.IOException;
+
+import org.apache.commons.vfs2.FileObject;
+import org.apache.commons.vfs2.impl.DefaultFileSystemManager;
+import org.apache.log4j.BasicConfigurator;
+import org.apache.log4j.Level;
+import org.apache.log4j.Logger;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.objectweb.proactive.extensions.dataspaces.vfs.VFSFactory;
+import org.objectweb.proactive.extensions.dataspaces.vfs.selector.FileSelector;
+import org.objectweb.proactive.extensions.vfsprovider.FileSystemServerDeployer;
+import org.objectweb.proactive.extensions.vfsprovider.client.ProActiveFileObject;
+
+
+/**
+ * @author ActiveEon Team
+ * @since 10/07/2017
+ */
+public class ProActiveFileObjectTest {
+
+    @ClassRule
+    public static TemporaryFolder tempFolder = new TemporaryFolder();
+
+    static File createdFolder;
+
+    static FileSystemServerDeployer deployer;
+
+    static DefaultFileSystemManager manager;
+
+    static String proActiveFSURL;
+
+    @BeforeClass
+    public static void setUp() throws IOException {
+
+        BasicConfigurator.resetConfiguration();
+        BasicConfigurator.configure();
+        Logger.getLogger(ProActiveFileObject.class).setLevel(Level.TRACE);
+        createdFolder = tempFolder.newFolder();
+        deployer = new FileSystemServerDeployer(createdFolder.getAbsolutePath(), false);
+        for (String url : deployer.getVFSRootURLs()) {
+            if (!url.startsWith("file:")) {
+                proActiveFSURL = url;
+            }
+        }
+        manager = VFSFactory.createDefaultFileSystemManager();
+    }
+
+    @Test
+    public void testFindFiles() throws IOException {
+        int nbFiles = 5000;
+        File findFilesFolder = new File(createdFolder, "findFiles");
+        findFilesFolder.mkdirs();
+        for (int i = 0; i < nbFiles; i++) {
+            File fileToFind = new File(findFilesFolder, "file_" + i + ".in");
+            fileToFind.createNewFile();
+        }
+        FileObject rootFO = manager.resolveFile(proActiveFSURL);
+        FileObject[] answer = rootFO.findFiles(new FileSelector("findFiles/*.in"));
+        assertTrue(answer.length == nbFiles);
+
+    }
```

Comment:
The depth of this test is 1, correct? Maybe another test with several files at different depths would be useful.